### PR TITLE
chore(queue): don't store queue rule config in redis

### PR DESCRIPTION
**Pull request by sileht**
This is not needed anymore, we always have the last version of the
configuration everywhere now.

Fixes MRGFY-522

Change-Id: I8fe7fe9ad4baf1a4a6b82bfb29052c87247657e5

### CODE REVIEW

File: mergify_engine/queue/merge_train.py [hunk 1]
```diff
@@ -1154,6 +1149,20 @@ async def _split_failed_batches(self, queue_rules: rules.QueueRules) -> None:
...
                 )
+
+                queue_name = car.still_queued_embarked_pulls[0].config["name"]
+                try:
+                    queue_rule = queue_rules[queue_name]
+                except KeyError:
+                    # We just need to wait the pull request has been removed from
+                    # the queue by the action
+                    self.log.info(
+                        "can split failed batch TrainCar, queue rule does not exists anymore",
```
Comment (jd) [inline mergify_engine/queue/merge_train.py:1160]:
```suggestion
                        "cant split failed batch TrainCar, queue rule does not exist anymore",
```
not sure it's really clear though 😢

File: mergify_engine/queue/merge_train.py [hunk 2]
```diff
@@ -1228,7 +1233,23 @@ async def _populate_cars(self, queue_rules: rules.QueueRules) -> None:
...
             self._current_base_sha = await self.get_head_sha()
 
-        speculative_checks = head.config["queue_config"]["speculative_checks"]
+        try:
+            queue_rule = queue_rules[head.config["name"]]
+        except KeyError:
+            # We just need to wait the pull request has been removed from
+            # the queue by the action
+            self.log.info(
+                "can populate cars, queue rule does not exists",
```
Comment (jd) [inline mergify_engine/queue/merge_train.py:1242]:
```suggestion
                "cant populate cars, queue rule does not exist",
```

File: mergify_engine/queue/merge_train.py [hunk 3]
```diff
@@ -1228,7 +1233,23 @@ async def _populate_cars(self, queue_rules: rules.QueueRules) -> None:
...
+                queue_rules=queue_rules,
+                queue_name=head.config["name"],
+            )
+            car = TrainCar(self, [head], [head], [], self._current_base_sha)
+            await car._set_creation_failure(
+                f"queue named `{head.config['name']}` does not exists anymore"
```
Comment (jd) [inline mergify_engine/queue/merge_train.py:1248]:
```suggestion
                f"queue named `{head.config['name']}` does not exist anymore"
```

**Review by jd — CHANGES_REQUESTED** [review]

**Comment by sileht** [comment]
@mergifyio squash first-commit

**Review by jd — APPROVED** [review]

**Pull request merged**